### ISSUE TITLE
When *both* verify_api_cert and ssl_verify_mode options are set in knife.rb, it doesn't respect either

### ISSUE THREAD

**Issue by yvovandoorn** [opscode/chef#2538]
Scenario:
Using knife under chef-client 12 against a Chef server with a self signed SSL certificate requires out of box modifications to the knife.rb file to not throw an error.

When both options:
```ruby
verify_api_cert		 "false"
ssl_verify_mode 	 :verify_none
```

Are present in the knife.rb file, you will get errors like this:
```ruby
ERROR: SSL Validation failure connecting to host: manage-demo.opscode.us - SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
ERROR: OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
```

When I remove 'verify_api_cert', there is no longer an error.

**Comment by lamont-granquist** [comment]
The latest ChefDK still doesn't have Chef-12 installed by default as the knife command line tool, this will be fixed when ChefDK 0.4.0 is released.

**Comment by sam3k** [comment]
@lamont-granquist is there a work around? I tried all the suggestions above and nothing helped: (removed verify_api_cert false, changed to verify_api_cert "false", etc)

```bash
Running handlers:
[2016-04-29T19:38:15-04:00] ERROR: Running exception handlers
Running handlers complete
[2016-04-29T19:38:15-04:00] ERROR: Exception handlers complete
Chef Client failed. 19 resources updated in 290.021195548 seconds
[2016-04-29T19:38:16-04:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2016-04-29T19:38:16-04:00] ERROR: remote_file[/usr/local/src/phantomjs-1.9.2-linux-x86_64.tar.bz2] (phantomjs::source line 36) had an error: OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
[2016-04-29T19:38:16-04:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

My .chef/knive.rb looks like this:

```bash
cookbook_path [ '~/my-app/chef/cookbooks' ]

ssl_verify_mode :verify_none
verify_api_cert false
```
